### ISSUE TITLE
Link in the new (3.14.0) annotation module returns an empty dict.

### ISSUE THREAD

**Issue by ymareth** [py-pdf/pypdf#2058]
The new Link class in annotation returns an empty* dict and hence creates no link.  
I took a look at the link class and can't see any calls to self.update (since the base class is dict) as I do in the other annotation classes.

I reverted back to pypdf.generic.AnnotationBuilder.add_link  with the same call parameters and links are inserted as expected.

*possibly 'title' would get through via the kwargs into the super constructor.

## Environment

Windows, Python 3.11, running version 3.14.0

**Comment by pubpub-zz** [comment]
@ymareth 
can you please provide standalone test code

**Comment by ymareth** [comment]
For version 3.13.0 this produces a link...

```
import pypdf as pp
import pypdf.generic as ppg

writer = pp.PdfWriter()
writer.add_blank_page(720, 720)
writer.write('test.pdf')

writer = pp.PdfWriter()
with open('test.pdf' , 'rb') as fh:
    reader = pp.PypdfReader(fh)
    writer.add_page(reader.pages[0])
    link = ppg.AnnotationBuilder.link(rect=(50, 100, 150, 200), targer_page_index=1)
    print(link)

```
Results in what looks like a healthy link definition...
``` 
{'/Type': '/Annot', '/Subtype': '/Link', '/Rect': RectangleObject([50, 100, 150, 200]), '/Border': [0, 0, 0], '/Dest': {'target_page_index': 1, 'fit': '/Fit', ''fit_args': []}}
```

For version 3.14.0 we have to change the code a bit to use the annotations module..

```
import pypdf as pp
import pypdf.annotations as ppa

writer = pp.PdfWriter()
writer.add_blank_page(720, 720)
writer.write('test.pdf')

writer = pp.PdfWriter()
with open('test.pdf' , 'rb') as fh:
    reader = pp.PypdfReader(fh)
    writer.add_page(reader.pages[0])
    link = ppa.Link(rect=(50, 100, 150, 200), targer_page_index=1)
    print(link)

```

Results in and empty dict.

```
{}
```

**Comment by MartinThoma** [comment]
Thank you for reporting the issue. I've adjusted the first code sample a bit and will have a look.

**Comment by MartinThoma** [comment]
Found the issue. Fix is on its way.

**Comment by MartinThoma** [comment]
Thank you for letting us know :pray: 

The issue is fixed with `pypdf==3.15.0`

**Comment by MartinThoma** [comment]
If you want, I can add you as a contributor: https://pypdf.readthedocs.io/en/latest/meta/CONTRIBUTORS.html
